src/org/zalando/stups/kio/api.clj

```diff
@@ -84,10 +85,19 @@
 
 ;; applications
 
+;;https://github.com/clojure/core.memoize/blob/master/docs/Using.md#overriding-the-cache-keys
+(defn ^{:clojure.core.memoize/args-fn first}
+  run-db-query
+  [params fn db-spec]
```

Comment:
Please don't shadow `fn`.

Reply:
Instead of passing `fn` please put the function name in params, it would be easier to understand what are we caching on.